scripts/pre_commit_linter.py

```diff
@@ -342,10 +361,57 @@ def _pre_commit_linter():
     summary_messages.append(py_result.get())
     print '\n'.join(summary_messages)
     print ''
+    return summary_messages
+
+
+# This part is used for detecting bad patterns.
```

Comment:
Change this into a docstring too. Say what the function actually does; don't refer to "this part" -- the function shouldn't need to know anything about its context beyond the args which are passed into it.